sklearn/inspection/_permutation_importance.py

```diff
@@ -30,9 +40,31 @@ def _calculate_permutation_scores(
     # (memmap). X.copy() on the other hand is always guaranteed to return a
     # writable data-structure whose columns can be shuffled inplace.
     X_permuted = X.copy()
+    y_mod = y.copy()
```

Comment:
`y_mod` is not a super descriptive name :)

Do we need to make a copy here? Making the indexing will just trigger a copy, isn't it?